src/Native/Unix/System.Security.Cryptography.Native.Apple/pal_trust.c

```diff
@@ -165,57 +164,104 @@ static int32_t EnumerateTrust(SecTrustSettingsDomain domain,
 
 int32_t AppleCryptoNative_StoreEnumerateUserRoot(CFArrayRef* pCertsOut, int32_t* pOSStatusOut)
 {
-    if (pCertsOut != NULL)
-        *pCertsOut = NULL;
+    if (pCertsOut == NULL)
+    {
+        if (pOSStatusOut != NULL)
+            *pOSStatusOut = noErr;
+        return -1;
+    }
+
+    CFMutableArrayRef pCertsRef = NULL;
+    int32_t ret;
+    
+    ret = EnumerateTrust(kSecTrustSettingsDomainUser,
+                         kSecTrustSettingsResultTrustRoot,
+                         &pCertsRef,
+                         pOSStatusOut);
+
+    *pCertsOut = pCertsRef;
 
-    return EnumerateTrust(kSecTrustSettingsDomainUser,
-                          kSecTrustSettingsResultTrustRoot,
-                          pCertsOut,
-                          pOSStatusOut);
+    return ret;
 }
 
 int32_t AppleCryptoNative_StoreEnumerateMachineRoot(CFArrayRef* pCertsOut, int32_t* pOSStatusOut)
 {
-    if (pCertsOut != NULL)
-        *pCertsOut = NULL;
+    if (pCertsOut == NULL)
+    {
+        if (pOSStatusOut != NULL)
+            *pOSStatusOut = noErr;
+        return -1;
+    }
 
-    CFMutableArrayRef* pCertsRef = pCertsOut;
+    CFMutableArrayRef pCertsRef = NULL;
+    int32_t ret;
 
-    int32_t ret =
-        EnumerateTrust(kSecTrustSettingsDomainAdmin, kSecTrustSettingsResultTrustRoot, pCertsRef, pOSStatusOut);
+    ret = EnumerateTrust(kSecTrustSettingsDomainAdmin,
+                         kSecTrustSettingsResultTrustRoot,
+                         &pCertsRef,
+                         pOSStatusOut);
 
     if (ret == 1)
     {
-        ret = EnumerateTrust(kSecTrustSettingsDomainSystem, kSecTrustSettingsResultTrustRoot, pCertsRef, pOSStatusOut);
+        ret = EnumerateTrust(kSecTrustSettingsDomainSystem,
+                             kSecTrustSettingsResultTrustRoot,
+                             &pCertsRef,
+                             pOSStatusOut);
     }
 
+    *pCertsOut = pCertsRef;
+
     return ret;
 }
 
 int32_t AppleCryptoNative_StoreEnumerateUserDisallowed(CFArrayRef* pCertsOut, int32_t* pOSStatusOut)
 {
-    if (pCertsOut != NULL)
-        *pCertsOut = NULL;
+    if (pCertsOut == NULL)
+    {
+        if (pOSStatusOut != NULL)
+            *pOSStatusOut = noErr;
+        return -1;
+    }
```

Comment:
Why does this intro need to change at all?

If the early return is required, then a more preferred pattern would be

```C
if (pCertsOut != NULL)
    *pCertsOut = NULL;
if (pOSStatusOut != NULL)
    *pOSStatusOut = noErr;

if (pCertsOut == NULL)
    return -1;
```

(Like it is in `EnumerateTrust`)

Though, really, it looks like the answer here is to move the type change to EnumerateTrust, since it would reduce most of the change in this particular file down to one or two lines in that function.